### PR TITLE
JITM - show a link to settings after a module is activated

### PR DESCRIPTION
**Pull request by mikestottuk**
Allow a JITM that activates a module to be redirected to an admin page if activated rather than simply fade out.

## Proposed changes:
* This adds a new flow to redirect to settings if a JITM is clicked which has an activate module as the CTA. Previously this would just fade out after 2 seconds and leave people where they were.
* If nothing is passed (e.g. we don't have a line like `->settings_redirect( 'jetpack#/newsletter' )`) the JITM will fade away as previously.
* Review alongside D120595-code

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
See here: pbNhbs-88F-p2#comment-17900

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* See D120595-code for detailed testing instructions.

### CODE REVIEW

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (jboland88) [inline projects/packages/jitm/src/js/jetpack-jitm.js]: Something about this doesn't sit right.

From a UX perspective, it seems unexpected that clicking an "Activate" button would redirect the user IMO.

I also think redirecting with `window.location.href` by pulling from the `window` object and the DOM might come with some risks. It could leave the door open to redirect somewhere else if another script from a different plugin decided to do something unfriendly and manipulate the window and DOM. 

cc @IanRamosC for some additional thoughts/ second opinion on this redirect method.

Would it work to swap the activate button for a "settings" button after the activation is complete?

**Review by jboland88 — COMMENTED** [review]
I am not sure about this solution - It works, but it feels like a hack. I'm going to ask for a second opinion from the team.

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (mikestottuk) [reply projects/packages/jitm/src/js/jetpack-jitm.js]: > Would it work to swap the activate button for a "settings" button after the activation is complete?

Interesting - the main issue was the fade away without a clear next step. Swapping it to stay in place and have Activate => Settings would work. Then they can dismiss it with the "X" or click settings to go to where the re-direct would be. 

Let me modify on that basis.

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (jboland88) [inline projects/packages/jitm/src/js/jetpack-jitm.js]: Curious here, wouldn't this disable clicks on the updated settings button?

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (jboland88) [inline projects/packages/jitm/src/js/jetpack-jitm.js]: Nice improvement over the redirect 👍  

What would you think about passing the full settings URL for the activated module in the JITM response and adding a settings button in the JITM template? 
Then, when you get to this step, you could just hide the activation button and show the settings button.

This allows us to avoid splitting up the URL into two parts, one that comes from the plugin and one that comes from the JITM response and having to re-assemble here. It will likely be less fragile to have the entire settings URL coming from a single source.

**Review by jboland88 — CHANGES_REQUESTED** [review]
This is improved, I had one comment about having the entire settings url coming from the JITM response (similar to how we specify the CTA URL) instead of part of the URL coming from the plugin and then needing to assemble the URL in code.

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (mikestottuk) [reply projects/packages/jitm/src/js/jetpack-jitm.js]: Yeah was trying to test this bit and whether it sent me to the link, clicking settings without it started to try and activate the module again.

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (mikestottuk) [reply projects/packages/jitm/src/js/jetpack-jitm.js]: Yeah, I think I'll do this - have another button drawn if it has the data in the envelope and then show hide it with JS

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (IanRamosC) [reply projects/packages/jitm/src/js/jetpack-jitm.js]: >cc @IanRamosC for some additional thoughts/ second opinion on this redirect method.

Just saw the comment now (that this seems to be resolved already), but I wanted to leave a comment for posterity. I agree with you. It's risky to redirect that way, especially if the redirect URL comes from the `window` object or anything else we don't have control over unless we validate it.

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (jboland88) [inline projects/packages/jitm/src/js/jetpack-jitm.js]: If this is (in theory) always going to be used as part of the module activation routine, would it make sense to nest this logic in the check for `if ( envelope.active_module )` ?

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (jboland88) [inline projects/packages/jitm/src/js/jetpack-jitm.js]: I think we'd want to keep these lines in the case that the activation button did not have a settings button to show. Could these lines just be moved to after the new if statement you added?

**Review by jboland88 — CHANGES_REQUESTED** [review]
I think this is looking close - just a couple minor points of feedback.

**Review by jboland88 — DISMISSED** [review]
It looks like you might need to run the version update script. Otherwise, this LGTM!

**Review by robertsreberski — DISMISSED** [review]
Looks like dependencies are updated correctly 👍

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (jboland88) [inline projects/packages/jitm/src/js/jetpack-jitm.js]: We actually don't need the `jptracks` class on buttons with the `jitm-button` class anymore - these buttons are tracked from within this file now.

File: projects/packages/jitm/src/js/jetpack-jitm.js
Comment (jboland88) [inline projects/packages/jitm/src/js/jetpack-jitm.js]: Same here, could you remove `jptracks` to avoid duplicate tracking?

**Review by jboland88 — APPROVED** [review]
Re-approving after version management updates

**Pull request merged**
